plugins/ImageGraph/API.php

```diff
@@ -129,404 +130,453 @@ public function get(
-        }
-
-        $useUnicodeFont = array(
-            'am', 'ar', 'el', 'fa', 'fi', 'he', 'ja', 'ka', 'ko', 'te', 'th', 'zh-cn', 'zh-tw',
-        );
-        $languageLoaded = Translate::getLanguageLoaded();
-        $font = self::getFontPath(self::DEFAULT_FONT);
-        if (in_array($languageLoaded, $useUnicodeFont)) {
-            $unicodeFontPath = self::getFontPath(self::UNICODE_FONT);
-            $font = file_exists($unicodeFontPath) ? $unicodeFontPath : $font;
-        }
-
-        // save original GET to reset after processing. Important for API-in-API-call
-        $savedGET = $_GET;
-
-        try {
-            $apiParameters = array();
-            if (!empty($idGoal)) {
-                $apiParameters = array('idGoal' => $idGoal);
+        $idSiteArr = array();
+        if (strpos($idSite, ',') !== false) {
+            $idSiteList = explode(',', $idSite);
+            foreach ($idSiteList as $key) {
+                $idSiteArr[] = $key;
             }
-            // Fetch the metadata for given api-action
-            $parameters = array(
-                'idSite' => $idSite,
-                'apiModule' => $apiModule,
-                'apiAction' => $apiAction,
-                'apiParameters' => $apiParameters,
-                'language' => $languageLoaded,
-                'period' => $period,
-                'date' => $date,
-                'hideMetricsDoc' => false,
-                'showSubtableReports' => true
-            );
-
-            $metadata = Request::processRequest('API.getMetadata', $parameters);
-            if (!$metadata) {
-                throw new Exception('Invalid API Module and/or API Action');
+        } else if (!empty($idSite) && ($idSite != 'all')) {
+            $idSiteArr[] = $idSite;
+        } else if (!empty($idSite) && ($idSite == 'all')) {
+            $viewableIdSites = APISitesManager::getInstance()->getSitesIdWithAtLeastViewAccess();
+            foreach ($viewableIdSites as $site) {
+                $idSiteArr[] = $site;
             }
+        } else {
+            // empty or invalid idSite has been already handled somewhere
+            exit;
+        }
 
-            $metadata = $metadata[0];
-            $reportHasDimension = !empty($metadata['dimension']);
-            $constantRowsCount = !empty($metadata['constantRowsCount']);
-
-            $isMultiplePeriod = Period::isMultiplePeriod($date, $period);
-            if (!$reportHasDimension && !$isMultiplePeriod) {
-                throw new Exception('The graph cannot be drawn for this combination of \'date\' and \'period\' parameters.');
+        $pngImgArr = array();
+        // remove duplicated idSite; improves image graph generation time.
+        $idSiteArr = array_unique($idSiteArr);
+        $idSiteArrCount = count($idSiteArr);
+        for ($w = 0; $w < $idSiteArrCount; $w++)
+        {
+            $idSite = $idSiteArr[$w];
+            Piwik::checkUserHasViewAccess($idSite);
```

Comment:
Let's consider we have

```
<?php
$idSiteArr = array(1,2,3,4,5);
```

Is it possible to pass an array of idSites like this outside the for loop?

for example,
Piwik::checkUserHasViewAccess($idSiteArr);

I had used like this **"Piwik::checkUserHasViewAccess($idSite);"** because I thought this function could check idSite one at a time. Could it take array as given above?  

Plus what will happen if user doesn't have view access only for idSide=3. Will it output other 4 image graphs or just stop after outputting first 2?